src/components/Layout/Layout.test.jsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+import { useAuth } from '../../providers/Auth';
+import Layout from './Layout.component';
+
+const mockHistoryPush = jest.fn();
+const mockLogout = jest.fn();
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useHistory: () => ({
+    push: mockHistoryPush,
+  }),
+}));
+
+jest.mock('../../providers/Auth', () => ({
+  useAuth: jest.fn(() => ({ authenticated: false })),
+}));
+
+describe('Layout', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it('should renders', () => {
+    render(
+      <Layout>
+        <div id="content">Content</div>
+      </Layout>
+    );
+
+    const container = document.querySelector('#content');
+    expect(container).toBeTruthy();
+    expect(screen.getByTestId('layout-sidebar')).toBeTruthy();
+    expect(screen.getByTestId('header-menu')).toBeTruthy();
+  });
+
+  it('should renders with the sidebar closed', () => {
+    render(
+      <Layout>
+        <div id="content">Content</div>
+      </Layout>
+    );
+
+    const sidebarOpened = document.body.getElementsByClassName(
+      'ui vertical labeled icon ui overlay left thin visible sidebar menu'
+    )[0];
+    expect(sidebarOpened).toBeFalsy();
+  });
+
+  it('should opens the sidebar when clicking bar icon', () => {
+    render(
+      <Layout>
+        <div id="content">Content</div>
+      </Layout>
+    );
+
+    fireEvent.click(screen.getByTestId('header-bars-icon'));
+    const sidebarOpened = document.body.getElementsByClassName(
```

Comment:
same as above, this could be reachable via data test id attribute

Reply:
if you want to be sure it has exactly that list of class names, you can iterate over the `classList` prop from the resultant object